### PR TITLE
Add 3D editor gizmo icons for Decal, LightmapProbe and FogVolume

### LINKED ISSUE #57283: Vulkan: Decals can't be selected by clicking on them in the 3D editor viewport
### Describe the project you are working on

Testing Godot v4.0.dev.20211015.official [f113dc986]

### Describe the problem or limitation you are having in your project

It is possible to have many of decals in the scene and struggle to **select** and manipulate them in easy way using 3d view.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The decals will be selectable in 3d view.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When user clicks a decals bounding box, the decal will become active.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It cant be worked around with a few lines of script.

### Is there a reason why this should be core and not an add-on in the asset library?

It is core functionality.

### LINKED ISSUE #78432: LightmapProbes not selectable in the 3D editor viewport
### Describe the project you are working on

N/A

### Describe the problem or limitation you are having in your project

As of currently, you cannot click-select them in 3D view and have to select them from scene tree. Considering that manually placing down light probes is already somewhat laborious(and using something to place them on a subdivided grid may not be enough), not being able to click-select them would be a huge pain.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It helps speed up the process of manually placing `LightmapProbe`s and moving them about. A huge QoL improvement in my opinion.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When the user clicks on a `LightmapProbe`'s gizmo(see the circular thing below), the `LightmapProbe` it belongs to will be selected and moved around easily. That's literally it.
image

### If this enhancement will not be used often, can it be worked around with a few lines of script?

`LightmapProbe` is implemented in core.

### Is there a reason why this should be core and not an add-on in the asset library?

See above.

### PR DESCRIPTION
**Pull request by Calinou**
- Tweak CPUParticles3D editor gizmo icon to use the same outline color as other gizmo icons. Make the internal icon easier to see by using an orange color instead of gray.

Lightmap probes that are automatically generated by LightmapGI don't have gizmo icons, as these are not nodes in the first place. Only manually authored LightmapProbe nodes have an icon.

## Preview

*Left: Decal, middle left: CPUParticles3D, middle right: GPUParticles3D, right: LightmapProbe*

image

*FogVolume:*

image

- This closes https://github.com/godotengine/godot/issues/57283 and closes https://github.com/godotengine/godot/issues/78432, as giving the decals and lightmap probes a gizmo icon makes them selectable by clicking the gizmo icon in the 3D editor viewport. (The same also applies to FogVolume.)

### CODE REVIEW

**Comment by jcostello** [comment]
@Calinou does it make it selectable with box select too?

**Comment by Calinou** [comment]
> @Calinou does it make it selectable with box select too?

Yes :slightly_smiling_face:

File: editor/icons/GizmoDecal.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg height="128" viewBox="0 0 128 128" width="128" xmlns="http://www.w3.org/2000/svg"><g stroke-width="8.00001"><path d="m111.31322 55.933581a48.000003 48.000002 0 1 0 -71.280012 49.527999z" fill="#f7f5cf"/><path d="m111.31322 55.933581a48.000003 48.000002 0 0 0 -71.280012 49.527999z" fill="#e1b44c"/><path d="m40.033208 105.46158a48.000003 48.000002 0 0 0 71.280012-49.527999z" fill="#f7f5cf" fill-opacity=".4"/></g><ellipse cx="64" cy="64" rx="50" ry="50.000004" style="fill:none;stroke:#000;stroke-width:3.99999;stroke-linecap:round;stroke-linejoin:round;stroke-opacity:.3"/></svg>
```
Comment (YuriSizov) [inline editor/icons/GizmoDecal.svg:1]: We should avoid `style` attributes, as they cannot be converted according to our colormap.

File: editor/icons/GizmoFogVolume.svg
Comment (YuriSizov) [inline editor/icons/GizmoFogVolume.svg:1]: This one also looks pretty big for what it is.

File: editor/plugins/gizmos/decal_gizmo_plugin.cpp [hunk 2]
```diff
@@ -38,6 +38,8 @@
 DecalGizmoPlugin::DecalGizmoPlugin() {
 	Color gizmo_color = EDITOR_DEF("editors/3d_gizmos/gizmo_colors/decal", Color(0.6, 0.5, 1.0));
 
+	create_icon_material("decal_icon", Node3DEditor::get_singleton()->get_editor_theme_icon(SNAME("GizmoDecal")));
```
Comment (YuriSizov) [inline editor/plugins/gizmos/decal_gizmo_plugin.cpp:41]:
```suggestion
	create_icon_material("decal_icon", EditorNode::get_singleton()->get_editor_theme()->get_icon(SNAME("GizmoDecal"), EditorStringName(EditorIcons)));
```

And others should be similar.

File: editor/icons/GizmoCPUParticles3D.svg
[the remark this answers is not in the document]
Comment (Calinou) [reply editor/icons/GizmoCPUParticles3D.svg:1]: Yes, I ran `svgcleaner --multipass GizmoCPUParticles3D.svg{,}` on it.

File: editor/icons/GizmoFogVolume.svg
Comment (Calinou) [reply editor/icons/GizmoFogVolume.svg:1]: I had to split the path to its own object to get it to render correctly, as ThorVG doesn't seem to behave well with outline drawing order properties. (I need the translucent outline to always draw *behind* the fill.)

File: editor/icons/GizmoDecal.svg (hunk 1 above)
Comment (Calinou) [reply editor/icons/GizmoDecal.svg:1]: Gizmo icons never have their colors converted (they are part of the 3D viewport which doesn't follow dark/light theme). I'd still like to get rid of those properties if possible, but I'm not sure how.

File: editor/plugins/gizmos/decal_gizmo_plugin.cpp (hunk 2 above)
Comment (Calinou) [reply editor/plugins/gizmos/decal_gizmo_plugin.cpp:41]: Should I also convert all existing icon materials not added by this PR?

File: editor/icons/GizmoCPUParticles3D.svg
[the remark this answers is not in the document]
Comment (YuriSizov) [reply editor/icons/GizmoCPUParticles3D.svg:1]: Maybe we need help from @MewPurPur then 🙂

File: editor/plugins/gizmos/decal_gizmo_plugin.cpp (hunk 2 above)
Comment (YuriSizov) [reply editor/plugins/gizmos/decal_gizmo_plugin.cpp]: > Should I also convert all existing icon materials not added by this PR?

No, that's already handled by https://github.com/godotengine/godot/pull/81516

File: editor/icons/GizmoFogVolume.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...70891.000005-7.999995 3.629109-8 8 .000005 4.370891 3.629109 7.999995 8 8h-8c-4.370891.000005-7.999995 3.629109-8 8 .000005 4.37089 3.629109 7.99999 8 8h16c-4.370891.00001-7.999995 3.62911-8 8 .000005 4.37089 3.629109 7.99999 8 8h40c4.370891-.00001 7.999995-3.62911 8-8-.000005-4.37089-3.629109-7.99999-8-8h-16c4.370891-.00001 7.999995-3.62911 8-8-.000005-4.370891-3.629109-7.999995-8-8h16c-4.370891.000005-7.999995 3.629109-8 8 .000005 4.37089 3.629109 7.99999 8 8h20c4.37089-.00001 7.99999-3.62911 8-8-.00001-4.370891-3.62911-7.999995-8-8h-4c4.37089-.000005 7.99999-3.629109 8-8-.00001-4.370891-3.62911-7.999995-8-8z" fill-opacity=".3"/><path d="m36 76a4.0000048 4.0000048 0 0 0 -4 4 4.0000048 4.0000048 0 0 0 4 4h64a4.0000048 4.0000048 0 0 0 4-4 4.0000048 4.0000048 0 0 0 -4-4zm-8 16a4.0000048 4.0000048 0 0 0 -4 4 4.0000048 4.0000048 0 0 0 4 4h40a4.0000048 4.0000048 0 0 0 4-4 4.0000048 4.0000048 0 0 0 -4-4zm56 0a4.0000048 4.0000048 0 0 0 -4 4 4.0000048 4.0000048 0 0 0 4 4h20a4.0000048 4.0000048 0 0 0 4-4 4.0000048 4.0000048 0 0 0 -4-4zm-40 16a4.0000048 4.0000048 0 0 0 -4 4 4.0000048 4.0000048 0 0 0 4 4h40a4.0000048 4.0000048 0 0 0 4-4 4.0000048 4.0000048 0 0 0 -4-4z" fill="#f7f5cf"/></svg>
```
Comment (MewPurPur) [inline editor/icons/GizmoFogVolume.svg:1]:
```suggestion
<svg height="128" viewBox="0 0 128 128" width="128" xmlns="http://www.w3.org/2000/svg"><path d="M20 68S8 68 8 56s16-12 16-12 0-32 28-32 28 24 28 24 10.08-16 24-8 4 24 4 24 12 0 12 8-8 8-8 8zm16 8a4 4 0 0 0 0 8h64a4 4 0 0 0 0-8zm-8 16a4 4 0 0 0 0 8h40a4 4 0 0 0 0-8zm56 0a4 4 0 0 0 0 8h20a4 4 0 0 0 0-8zm-40 16a4 4 0 0 0 0 8h40a4 4 0 0 0 0-8z" stroke="#000" stroke-width="8" stroke-opacity=".3"/><path d="M20 68S8 68 8 56s16-12 16-12 0-32 28-32 28 24 28 24 10.08-16 24-8 4 24 4 24 12 0 12 8-8 8-8 8zm16 8a4 4 0 0 0 0 8h64a4 4 0 0 0 0-8zm-8 16a4 4 0 0 0 0 8h40a4 4 0 0 0 0-8zm56 0a4 4 0 0 0 0 8h20a4 4 0 0 0 0-8zm-40 16a4 4 0 0 0 0 8h40a4 4 0 0 0 0-8z" fill="#f7f5cf"/></svg>
```

File: editor/icons/GizmoLightmapProbe.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...28125 33.328125s-14.874203 33.328125-33.328125 33.328125h-4v26.671875h4c33.089693 0 60-26.910331 60-60s-26.910307-60-60-60zm8 8.201172c26.8831 2.042253 48 24.383393 48 51.798828s-21.1169 49.75658-48 51.79883v-10.875c20.881668-2.0478 37.32813-19.515818 37.32813-40.92383s-16.446462-38.876031-37.32813-40.923828zm-32.279297 6.541015-16.978516 16.978516 2.830079 2.828125 13.460937 13.451172h-31.033203v24h4 27.033203l-16.292969 16.279297 2.832032 2.830078 14.148437 14.138675 16.96875-16.968753-16.689453-16.6875v-11.591797-11.591797l16.6875-16.6875zm.279297 45.257813c0 15.416587 12.583425 28 28 28s28-12.583413 28-28-12.583425-28-28-28-28 12.583413-28 28zm-.28125-33.941406 5.658203 5.660156-5.658203 5.658203-5.660156-5.658203zm28.28125 13.941406c11.093066 0 20 8.906927 20 20s-8.906934 20-20 20-20-8.906927-20-20 8.906934-20 20-20zm-52 16h16v8h-16zm23.71875 26.623047 5.65625 5.65625-5.65625 5.65625-5.660156-5.65625z" fill-opacity=".3"/><path d="m8 72h24v-16h-24zm16.4 20.28 11.32 11.312 11.312-11.312-11.312-11.312zm0-56.56 11.32 11.312 11.312-11.312-11.312-11.32zm15.6 28.28a24 24 0 0 0 48 0 24 24 0 0 0 -48 0zm24 56a56 56 0 0 0 0-112v18.672a37.328 37.328 0 0 1 0 74.656z" fill="#f7f5cf"/></svg>
```
Comment (MewPurPur) [inline editor/icons/GizmoLightmapProbe.svg:1]:
```suggestion
<svg height="128" viewBox="0 0 128 128" width="128" xmlns="http://www.w3.org/2000/svg"><path d="M8 72h24V56H8zm16.4 20.28 11.32 11.312L47.032 92.28 35.72 80.968zm0-56.56 11.32 11.312L47.032 35.72 35.72 24.4zM40 64a24 24 0 0 0 48 0 24 24 0 0 0-48 0zm24 56A56 56 0 0 0 64 8v18.672a37.328 37.328 0 0 1 0 74.656z" stroke="#000" stroke-width="8" stroke-opacity=".3"/><path d="M8 72h24V56H8zm16.4 20.28 11.32 11.312L47.032 92.28 35.72 80.968zm0-56.56 11.32 11.312L47.032 35.72 35.72 24.4zM40 64a24 24 0 0 0 48 0 24 24 0 0 0-48 0zm24 56A56 56 0 0 0 64 8v18.672a37.328 37.328 0 0 1 0 74.656z" fill="#f7f5cf"/></svg>
```

**Comment by YuriSizov** [comment]
@MewPurPur You're brilliant! 🚀

File: editor/plugins/gizmos/decal_gizmo_plugin.cpp (hunk 2 above)
Comment (Calinou) [reply editor/plugins/gizmos/decal_gizmo_plugin.cpp]: Alright, I've converted the icon materials for icons added by this PR.

**Comment by Calinou** [comment]
@MewPurPur Thanks a lot! I've added you as a co-author :slightly_smiling_face:

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**